---
output:
  rmarkdown::html_document:
    highlight: pygments
    toc: true
    toc_depth: 3
    fig_width: 5
bibliography: "`r system.file(package='MicrobiomeWorkshop', 'vignettes', 'bibliography.bib')`"
vignette: >
  %\VignetteIndexEntry{MicrobiomeWorkshop}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding[utf8]{inputenc}
---


## Working with open-source Human Microbiome Project Data phases 1 (HMP) and 2 (iHMP): Efficient Data Access and Analysis Workflow

## Instructors names and contact information

1. Levi Waldron (@LeviWaldron1), Graduate School of Public Health and Health Policy, City University of New York, New York, NY. 
email: Levi.Waldron@sph.cuny.edu

2. Ni Zhao, Department of Biostatistics, Johns Hopkins University, Baltimore, MD. 
email: nzhao10@jhu.edu

3. Ekaterina Smirnova (@katiasmirn), Department of Biostatistics, Virginia Commonwealth University, Richmond, VA. email: ekaterina.smirnova@vcuhealth.org

## Workshop Description

The composition of microbial species in a human body is essential for maintaining human health, and it is associated with a number of diseases including obesity, bowel inflammatory disease, and bacterial vaginosis.  Over the last decade, microbiome data analysis  almost entirely shifted towards using samples taken directly from various sites of human body and  to explore a  large number of microbes using 16S or whole metagenome sequencing. This technological shift has led to a radical change  in the data collected and led to the creation  of the  Human Microbiome Project (HMP) in 2008. With the growth and success of the microbiomics field, the size and complexity, and availability of the microbiome data in any given experiment have increased exponentially.  Publicly-available data sets from amplicon sequencing of two 16S ribosomal RNA variable regions, with extensive controlled-access participant data, provide a reference for ongoing microbiome studies. However, utilization of these data sets can be hindered by the complex bioinformatic steps required to access, import, decrypt, and merge the various components in formats suitable for ecological and statistical analysis. 

The main goal of this workshop is to provide a much-needed tutorial on downloading, understanding, and analyzing  the publicly-available HMP data. We will describe the two packages, HMP16SData and HMP2Data, that provide the comprehensive tutorial on the analysis of Phase 1 and 2 of the HMP project, respectively. These packages provide count data for both 16S ribosomal RNA variable regions, integrated with phylogeny, taxonomy, public participant data, and controlled participant data for authorized researchers, using standard integrative Bioconductor data objects.  As such, these packages provide the first comprehensive publicly-available tool for working with  microbiome data collected at different body sites (e.g., fecal, nasal, vaginal) coupled with other omics approaches (e.g., cytokines, whole metagenome), and is conducted longitudinally on multiple subjects at multiple visits. By removing bioinformatic hurdles of data access and management, HMP16SData and HMP2Data  enable researchers with only basic R skills to quickly analyze HMP data. This allows  studying the dynamics of microbial composition over the disease progression, discovering  disease-specific microbial biomarkers, and understanding the role of  microbiome in connection to other omics measurements.  

This workshop will consist of the series  on the instructors-led live demos presented together with  the brief lecture materials and statistical methods review when necessary. Completely worked out examples are available through the HMP16SData and HMP2Data package vignettes. 



### Pre-requisites

* Basic knowledge of R syntax
* Familiarity with the microbiome data 
* Familiarity with alpha and beta diversity in the context of microbiome data analysis
* Familiarity with principal components and principal coordinates analysis

Background reading:

* Schiffer L, Azhar R, Shepherd L, Ramos M, Geistlinger L, Huttenhower C, Dowd JB, Segata N, Waldron L (2019). "HMP16SData: Efficient Access to the Human Microbiome Project through Bioconductor." American Journal of Epidemiology. doi: 10.1093/aje/kwz006.



### Workshop Participation

Workshop participants are expected to bring a lap top, download HMP16SData and HMP2Data packages in advance, and be able to execute  R commands throughout workshop. 

### _R_ / _Bioconductor_ packages used

HMP16SData

HMP2Data

### Time outline 

| Activity                     | Time |
|------------------------------|------|
| HMP16SData                   | 10m  |
| HMP2Data                     | 10m  |
| Microbiome Analyses          | 15m  |
| Multi-omics data integration | 15m  |

## Workshop goals and objectives

The main purpose of this workshop is to help researchers interested in public microbiome data to start using  the HMP16SData and HMP2Data packages.  

### Learning goals

* understand the goals of  HMP projects and the studies available through the HMP DAC portal 
* download and use HMP data packages
* understand the difference between phase 1 and phase 2 projects
* understand principles of -omics data integration and visualization

### Learning objectives

* install HMP data packages and access vignettes
* download HMP data to produce analyses illustrated in package vignettes
* create sample summary statistics
* create alpha and beta diversity plots
* understand longitudinal patterns and examine microbiome diversity changes  
* integrate 16S and cytokines data using co-inertia analyses techniques 
* learn capturing patterns across -omics data sets of different structure

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Install packages

Installing all necessary packages can be accomplished by the following command:

```{r, eval=FALSE}
library(BiocManager)
install("waldronlab/MicrobiomeWorkshop", build_vignettes=TRUE, 
        dependencies=TRUE)
```

Then you should be able to view the compiled vignette:
```{r, eval=FALSE}
vignette("MicrobiomeWorkshop")
```


Load all needed packages. These are listed in the [DESCRIPTION](https://github.com/waldronlab/MicrobiomeWorkshop/blob/master/DESCRIPTION) file: currently 
```{r, eval=FALSE}
library(MicrobiomeWorkshop)
```

## Introduction

Bioconductor provides curated resources of microbiome data.  Most microbiome data are generated either by targeted amplicon sequencing (usually of variable regions of the 16S ribosomal RNA gene) or by metagenomic shotgun sequencing (MGX). These two approaches are analyzed by different sequence analysis tools, but downstream statistical and ecological analysis can involve any of the following types of data:

* taxonomic abundance at different levels of the taxonomic hierarchy
* phylogenetic distances and the phylogenetic tree of life
* metabolic potential of the microbiome
* abundance of microbial genes and gene families

A review of types and properties of microbiome data is provided by [@Morgan2012-mq].

### curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub

```{r, results="hide", echo=FALSE, cache=FALSE}
suppressPackageStartupMessages(library(curatedMetagenomicData))
```

`curatedMetagenomicData`[@Pasolli2017-gf] provides 6 types of processed data for >30 publicly available whole-metagenome shotgun sequencing datasets, including from the Human Microbiome Project Phase 1:

1. Species-level taxonomic profiles, expressed as relative abundance from kingdom to strain level
2. Presence of unique, clade-specific markers
3. Abundance of unique, clade-specific markers
4. Abundance of gene families
5. Metabolic pathway coverage
6. Metabolic pathway abundance

Types 1-3 are generated by 
[MetaPhlAn2](http://huttenhower.sph.harvard.edu/metaphlan2); 4-6 are generated by [HUMAnN2](http://huttenhower.sph.harvard.edu/humann2).

Currently, `curatedMetagenomicData` provides:

* `r nrow(combined_metadata)` samples from `r length(unique(combined_metadata$dataset_name))` datasets, primarily of the human gut but including body sites profiled in the Human Microbiome Project
* Processed data from whole-metagenome shotgun metagenomics, with manually-curated metadata, as integrated and documented Bioconductor 
ExpressionSet objects
* ~80 fields of specimen metadata from original papers, supplementary files, and websites, with manual curation to standardize annotations
* Processing of data through the [MetaPhlAn2](http://huttenhower.sph.harvard.edu/metaphlan2) pipeline for taxonomic abundance, and [HUMAnN2](http://huttenhower.sph.harvard.edu/humann2) pipeline for metabolic analysis
* These represent ~100TB of raw sequencing data, but the processed data provided are much smaller.

These datasets are documented in the 
[reference manual](https://bioconductor.org/packages/devel/data/experiment/manuals/curatedMetagenomicData/man/curatedMetagenomicData.pdf).

This is an `ExperimentHub` package, and its main workhorse function is `curatedMetagenomicData()`:

The manually curated metadata for all available samples are provided in a single table `combined_metadata`:

```{r eval=FALSE}
library(curatedMetagenomicData)
?combined_metadata
View(data.frame(combined_metadata))
```

The main function provides a `list` of `ExpressionSet` objects:
```{r, results="hide"}
oral <- c("BritoIL_2016.metaphlan_bugs_list.oralcavity",
          "Castro-NallarE_2015.metaphlan_bugs_list.oralcavity")
esl <- curatedMetagenomicData(oral, dryrun = FALSE)
```

```{r}
esl
```

These `ExpressionSet` objects can also be converted to `phyloseq` object for ecological analysis and differential abundance analysis using the `DESeq2` package, using the `ExpressionSet2phyloseq()` function:
```{r}
ExpressionSet2phyloseq( esl[[1]], phylogenetictree = TRUE)
```

See the documentation of `phyloseq` for more on ecological and differential abundance analysis of the microbiome.

### HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project

```{r, results='hide'}
suppressPackageStartupMessages(library(HMP16SData))
```

`HMP16SData`[@Schiffer2019-cd] is a Bioconductor ExperimentData package of the Human Microbiome Project (HMP) 16S rRNA sequencing data. Taxonomic count data files are provided as downloaded from the HMP Data Analysis and Coordination Center from its QIIME pipeline. Processed data is provided as `SummarizedExperiment` class objects via `ExperimentHub`. Like other ExperimentHub-based packages, a convenience function does downloading, automatic local caching, and serializing of a Bioconductor data class. This returns taxonomic counts from the V1-3 variable region of the 16S rRNA gene, along with the unrestricted participant data and phylogenetic tree. 
```{r}
V13()
```

This can also be converted to `phyloseq` for ecological and differential abundance analysis; see the `HMP16SData` vignette for details.


## How to add figures

`r knitr::include_graphics(system.file(package='MicrobiomeWorkshop', 'vignettes', 'fig.png'))`

